sematic/scheduling/kubernetes.py

```diff
@@ -517,41 +286,217 @@ def refresh_job(job: ExternalJob) -> KubernetesExternalJob:
         )
         if ts is None:
             logger.warning(
-                "Setting has_started=True for job %s with no start time!",
-                job.external_job_id,
+                "Setting has_started=True for job %s/%s with no start time!",
+                job.namespace,
+                job.name,
             )
         else:
             logger.info(
-                "Setting has_started=True for job %s at timestamp %s",
-                job.external_job_id,
+                "Setting has_started=True for job %s/%s at timestamp %s",
+                job.namespace,
+                job.name,
                 ts,
             )
-        job.has_started = True
-        job.start_time = ts
+        details.has_started = True
+        details.start_time = ts
 
-    # trust the status.active field over the state of pods,
-    # as explained in KubernetesExternalJob.is_active()
-    job.pending_or_running_pod_count = (
+    # Trust the status.active field over the state of pods. According to the docs:
+    # github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1JobStatus.md
+    # a job's "active" field holds the number of pending or running pods.
+    # This should be a more reliable measure of whether the job is still
+    # active than the number of succeeded or failed pods, as during pod
+    # evictions (which don't stop the job completely, but do stop the pod),
+    # a pod can briefly show up as failed even when another one is
+    # going to be scheduled in its place.
+    details.pending_or_running_pod_count = (
         k8s_job.status.active if k8s_job.status.active is not None else 0  # type: ignore
     )
-    job.succeeded_pod_count = (
+    details.succeeded_pod_count = (
         k8s_job.status.succeeded  # type: ignore
         if k8s_job.status.succeeded is not None  # type: ignore
         else 0
     )
 
-    job.most_recent_condition = _get_most_recent_job_condition(job, k8s_job)
-    (
-        job.most_recent_pod_phase_message,
-        job.most_recent_pod_condition_message,
-        job.most_recent_container_condition_message,
-        job.has_infra_failure,
-    ) = _get_most_recent_pod_details(job)
+    job.details = details
+
+    job.update_status(details.get_status(time.time()))
 
-    logger.debug("Job %s refreshed: %s", job.external_job_id, job)
+    logger.debug("Job %s/%s refreshed: %s", job.namespace, job.name, job)
```

Comment:
This usage pattern pops up repeatedly, Please implement these methods on `Job` and propagate usages:
```suggestion
    logger.debug("Job %s refreshed: %s", str(job), repr(job))
```

Reply:
done